### PR TITLE
Damage

### PR DESCRIPTION
**Pull request by Sorseg**

### CODE REVIEW

File: crates/petri_client/src/hud.rs [hunk 1]
```diff
@@ -71,3 +107,78 @@ fn receive_damage(
...
+    };
+    for (label_entity, mut style, &NameLabelFor(entity)) in &mut labels {
+        let Ok(entity_transform) = transforms.get(entity) else {
+            // no entity pos, then label is outdated
+            commands.entity(label_entity).despawn_recursive();
+            continue;
+        };
+        let transform = camera.world_to_viewport(camera_transform, entity_transform.translation());
+        let Some(transform) = transform else {
+            // label is not visible
```
Comment (demoth) [inline crates/petri_client/src/hud.rs:176]: what exactly means 'visible' here? Is the case when the label is outside the viewport is handled here?

File: crates/petri_client/src/plugin.rs [hunk 2]
```diff
@@ -39,251 +35,180 @@ pub enum PetriState {
...
+    let delta = mouse_motion_events.read().map(|e| e.delta).sum::<Vec2>();
+    // FIXME: limit X turn angle
+    transform.rotate_y(-delta.x * sensitivity);
+    transform.rotate_local_x(-delta.y * sensitivity);
+    events.send(Aim(transform.forward()));
+}
 
-            let current_time = SystemTime::now().duration_since(SystemTime::UNIX_EPOCH)?;
-            let client_id = current_time.as_millis() as u64;
+// Player id of the player who is playing this instance of the game
```
Comment (demoth) [inline crates/petri_client/src/plugin.rs:164]:
```suggestion
// Id of the player who is playing this instance of the game
```

Comment (demoth) [reply crates/petri_client/src/plugin.rs:164]: // Generated from the connection time in ms

File: crates/petri_server/src/enemy.rs [hunk 3]
```diff
@@ -21,14 +22,20 @@ pub struct EnemyPlugin;
...
+        );
     }
 }
 
 #[derive(Component)]
 pub(crate) struct Monster;
 
+/// After attacking, monster needs time to recover
+#[derive(Component)]
+struct Stun(Timer);
```
Comment (demoth) [inline crates/petri_server/src/enemy.rs:37]: AttackDelay?

Comment (demoth) [reply crates/petri_server/src/enemy.rs:37]: Or simply `Delay`, if we will reuse this component for other systems

**Review by demoth — APPROVED** [review]

File: crates/petri_client/src/hud.rs [hunk 4]
```diff
@@ -71,3 +107,78 @@ fn receive_damage(
         *h_bg = Color::RED.with_a(*fadeout).into();
     }
 }
+
+/// Component of the label entity that points to which 3d world entity this label is for
+#[derive(Component)]
+struct NameLabelFor(Entity);
+
+/// Component of a 3d world entity that points to the label entity for this entity
```
[the remark this answers is not in the document]
Comment (Sorseg) [reply crates/petri_client/src/hud.rs:115]: Yes

File: crates/petri_client/src/plugin.rs (hunk 2 above)
Comment (Sorseg) [reply crates/petri_client/src/plugin.rs]: I think this is an implementation detail that is likely to change, so little value of documenting it here

File: crates/petri_client/src/hud.rs (hunk 1 above)
Comment (Sorseg) [reply crates/petri_client/src/hud.rs:176]: > Is the case when the label is outside the viewport is handled here?

Yes and a [couple more edge cases](
https://docs.rs/bevy/0.13.1/bevy/render/camera/struct.Camera.html#method.world_to_viewport):


> Returns None if any of these conditions occur:
> - The computed coordinates are beyond the near or far plane
> - The logical viewport size cannot be computed. See [logical_viewport_size](https://docs.rs/bevy/0.13.1/bevy/render/camera/struct.Camera.html#method.logical_viewport_size)
> - The world coordinates cannot be mapped to the Normalized Device Coordinates. See [world_to_ndc](https://docs.rs/bevy/0.13.1/bevy/render/camera/struct.Camera.html#method.world_to_ndc) May also panic if glam_assert is enabled. See [world_to_ndc](https://docs.rs/bevy/0.13.1/bevy/render/camera/struct.Camera.html#method.world_to_ndc).

File: crates/petri_server/src/enemy.rs (hunk 3 above)
Comment (Sorseg) [reply crates/petri_server/src/enemy.rs:37]: I would change that once we have some more complex behavior, then we might just even delete that

**Pull request merged**